tests/integration/dhcp_test.py

```diff
@@ -192,6 +193,126 @@ def test_dhcp_for_bond_with_ip_address_and_slave(dhcp_env,
+    dhcp_cli_desired_state['ipv4']['dhcp'] = True
+    dhcp_cli_desired_state['ipv4']['ignore-auto-routes'] = True
+    dhcp_cli_desired_state['ipv6']['enabled'] = True
+
+    netapplier.apply(desired_state)
+    assertlib.assert_state(desired_state)
+    # TODO(Gris Ge): Once libnmstate support route query, need to make
+    # sure dhcp route got ignored. And this also require change DHCP server
+    # from dnsmasq to dhcpcd to pushing extra routes.
+
+
+def test_ipv6_dhcp_and_autoconf_ignore_gateway(dhcp_env):
+    desired_state = statelib.show_only((DHCP_CLI_NIC,))
+    dhcp_cli_desired_state = desired_state[INTERFACES][0]
+    dhcp_cli_desired_state['state'] = 'up'
+    dhcp_cli_desired_state['ipv6']['dhcp'] = True
+    dhcp_cli_desired_state['ipv6']['autoconf'] = True
+    dhcp_cli_desired_state['ipv6']['ignore-auto-gateway'] = True
+
+    netapplier.apply(desired_state)
+
+    assertlib.assert_state(desired_state)
+    # TODO(Gris Ge): Once libnmstate support route query, need to make
+    # sure DHCP gateway got ignored.
+
+
+def test_ipv6_dhcp_and_autoconf_ignore_dns(dhcp_env):
+    desired_state = statelib.show_only((DHCP_CLI_NIC,))
+    dhcp_cli_desired_state = desired_state[INTERFACES][0]
+    dhcp_cli_desired_state['state'] = 'up'
+    dhcp_cli_desired_state['ipv6']['dhcp'] = True
+    dhcp_cli_desired_state['ipv6']['autoconf'] = True
+    dhcp_cli_desired_state['ipv6']['ignore-auto-dns'] = True
+
+    netapplier.apply(desired_state)
+
+    assertlib.assert_state(desired_state)
+    # TODO(Gris Ge): Once libnmstate support DNS query, need to make
+    # sure DHCP DNS entires got ignored.
+
+
+def test_ipv6_dhcp_and_autoconf_ignore_routes(dhcp_env):
+    desired_state = statelib.show_only((DHCP_CLI_NIC,))
+    dhcp_cli_desired_state = desired_state[INTERFACES][0]
+    dhcp_cli_desired_state['state'] = 'up'
+    dhcp_cli_desired_state['ipv6']['dhcp'] = True
+    dhcp_cli_desired_state['ipv6']['autoconf'] = True
+    dhcp_cli_desired_state['ipv6']['ignore-auto-routes'] = True
+
+    netapplier.apply(desired_state)
+
+    assertlib.assert_state(desired_state)
+    # TODO(Gris Ge): Once libnmstate support route query, need to make
+    # sure dhcp route got ignored. And this also require change DHCP server
+    # from dnsmasq to dhcpcd to pushing extra routes.
+
+
+@pytest.mark.xfail(raises=validator.InvalidDynamicIpOption)
+def test_ipv4_dhcp_off_and_option_on(dhcp_env):
+    desired_state = statelib.show_only((DHCP_CLI_NIC,))
+    dhcp_cli_desired_state = desired_state[INTERFACES][0]
+    dhcp_cli_desired_state['state'] = 'up'
+    dhcp_cli_desired_state['ipv4']['dhcp'] = False
+    dhcp_cli_desired_state['ipv4']['ignore-auto-routes'] = True
+    dhcp_cli_desired_state['ipv4']['ignore-auto-dns'] = True
+    dhcp_cli_desired_state['ipv4']['ignore-auto-gateway'] = True
+    dhcp_cli_desired_state['ipv6']['enabled'] = True
+
+    netapplier.apply(desired_state)
+
+
+@pytest.mark.xfail(raises=validator.InvalidDynamicIpOption)
```

Comment:
`xfail` means it should pass but currently fails and you know about it.
In this case, you expect it to fail, so you should use `with pytest.raises`.

Reply:
Thanks for pointing that out. Will fix it soon.